### PR TITLE
Added page on setting up external editors

### PR DESCRIPTION
**Pull request by sdfgeoff**
Because I had to go and read the source to set it up properly.

There may be a better place to put it other than under scripting, so I'm open for suggestions.

### CODE REVIEW

File: getting_started/scripting/external_editor.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+There are two fields: the executable path and command line flags. The flags
+allow you to better integrate the editor with godot. Godot will replace the
+following inside the flags parameter:
+
++---------------------+-----------------------------------------------------+
+| Field in Exec Flags | Is replaced with                                    |
++=====================+=====================================================+
+| {project}           | The path to the project directory                   |
++---------------------+-----------------------------------------------------+
+| {file}              | The file path (reltaive to what?)                   |
```
Comment (neikeq) [inline getting_started/scripting/external_editor.rst]: Absolute paths both for the project and file.

**Comment by sdfgeoff** [comment]
Thanks. Updated.

**Comment by mhilbrunner** [comment]
There may be some overlap with PR https://github.com/godotengine/godot-docs/pull/1164

**Comment by Dar13** [comment]
This is very similar to #1164 (my PR). I was going to wait until #395 was resolved as documenting this feature is done as part of the proposed solution by @pwab. Could always merge this or #1164, and update it later to better fit the final solution.

**Comment by pwab** [comment]
> Could always merge this or #1164, and update it later to better fit the final solution.

Yeah that would be great. I won't be able to work on the settings overview pages before April and these two PRs are nice additions for the docs that should be merged as soon as possible.

**Comment by sdfgeoff** [comment]
Ahh, didn't notice #1164. Yes, definitely overlap. I'm with Dar13 on merging something and updating later, because this information is not available without reading the source, and cannot be figured out by experimentation.

As to who's PR, I don't particularly mind. Probably merge #1164 as it has a nicer introduction than mine.

**Comment by mhilbrunner** [comment]
I'll merge this one until @pwab's script is merged (https://github.com/godotengine/godot-docs/issues/395#issuecomment-364750829), we just need to remember to update this once this is automated.

**Pull request merged**